### DISCUSSION TITLE
Can't reach Web UI when devices connected to my home Wifi

### DISCUSSION THREAD

**Discussion by Go-rom** [mtotschnig/MyExpenses#1288] [Q&A] [answered]
Hi,

right now i'm using the F-Droid version of My Expenses with an extended key. (version 3.6.5.2)

I'm interested in the web user interface feature and before going with the professional key, i'm using the trial mode to be sure that this feature fits my need.

My android phone and my laptop are connected to the same home wifi network, i know that the web interface is on because i can see the ip address but i can't reach it from my laptop through the last stable version of Firefox.

Switching to https or not doesn't change anything. The same with a password or not.

Note that if i switch on the wifi hotspot on my phone then i connect my laptop to it, it works.

So i think i may need an extra setting on my wifi router, is that right ?

Thank you for this great app btw.

**Comment by mtotschnig** [comment]
The router must allow you to connect the phone and the laptop. If it doesn't maybe it has both of them on different sub-networks? Can you compare the IP addresses of both devices? Do they share the first three numbers, and only differ by the fourth number?

**Comment by Go-rom** [comment]
They are both on the same sub-network (they do share the first three numbers and only differ by the fourth number).
Could it be related to the fact that the server uses the port 9000 ?
FWIW, the IPv4 address provided by my ISP is not "full-stack", it only covers ports from 49152 to 65535 but i'm not able to say if it's relevant or not.

Just an additional information, i also tested with a secondary browser (Gnome Web) which has no extensions or fancy setting.

**Reply by mtotschnig** [reply]
I suppose that the IPv4 address provided by the ISP is the one of the router towards the outside. It should not have any relevance for the internal network that I suppose is in the 192.168.* range. My suggestion would be to check if any firewall is active on the router blocking traffic between devices on the internal network.

**Reply by Go-rom** [reply]
> I suppose that the IPv4 address provided by the ISP is the one of the router towards the outside.

That's right.
 
> that I suppose is in the 192.168.* range

Also right.

> My suggestion would be to check if any firewall is active on the router blocking traffic between devices on the internal network.

I just looked after anything which could block the traffic but no. But i think you're right, the wifi router provided by my ISP might block by design any local server.
Please leave this topic open for now, i'll run some researches and  i'll get back here with some (good or bad) news.
And i will try with another wifi router in order not to rely on the ISP's one.

Thank you.

**Reply by Go-rom** [reply]
Great news ! It's working thanks to Fric-Box from [ADUF boards](https://www.aduf.org/viewtopic.php?p=2002837#2002837) (french boards).

Here's the answer, when the Web server starts, My Expenses displays the following IP adresse : http(s)://10.1.10.1:9000
And the thing is the help section says that we have to call the URL this IP address from a web browser on the desktop computer.
That works when the mobile phone is used as a Wi-Fi hotspot but not when there's a Wi-Fi router between the phone and the computer.

When comes the case with the router, the called IP address has to be the one provided by the Wi-Fi router (in my case 192.168.0.X:9000).
I know that it makes sense but i think i was confused by the help text and the fact that i'm a rookie in that field.

Do you think that it would be relevant to enhance the help text with this information ?

**Reply by mtotschnig** [reply]
@Go-rom I am glad that you were able to work it out. I am surprised that My Expenses would not display the address provided by the Wi-Fi router, which it does on all devices I have tested with so far. Could you describe what type of device on which Android version you are using?

**Reply by Go-rom** [reply, marked as answer]
Hello @mtotschnig 

your last answer aroused my curiosity, i wasn't aware that My Expenses should display the address provided by the Wi-Fi router. (That's now makes sense)
So i thought that the issue had to be on my side and i digged up further.
I can tell you that i found the culprit : Tracker Control, an android free software app which blocks most of trackers embedded in some apps.
It uses the android VPN feature in order to monitor I/O net traffic. (i am forced to use the VPN feature because my devices is not rooted).

I am aware that My Expenses (F-Droid version) embeds no trackers but i have to admit that i don't take time to whitelist all my apps provided by F-Droid.

So whitelisting My Expenses solved the issue, i'm now able to get the right gateway IP address.
I would suggest to add a small note in the Web UI settings in order to warn users about that kind of app. 

Thank you for your help, the Web UI feature is great i just bought the professional key.
Also let me wish you all the best for 2024.
